File: JavaScript/src/GeneratorAlgorithm.js
function randomInt(min, max) {
  min = Math.ceil(min);
  max = Math.floor(max);
  return Math.floor(Math.random() * (max - min + 1)) + min;
}

class GeneratorAlgorithm {
  generateCPF() {
    this.buildCPF = [];

    for (let i = 0; i < 9; i++) {
      this.buildCPF.push(randomInt(0, 9));
    }

    for (let i = 0; i < 2; i++) {
      this.buildCPF.push(this.digit());
    }

    return this.buildString();
  }

  digit() {
    let sum = 0;
    let index = this.buildCPF.length + 1;

    for (let i = 0; i < this.buildCPF.length; i++) {
      sum += this.buildCPF[i] * index;
      index--;
    }

    return this.verifySum(sum);
  }

  verifySum(sum) {
    if (sum % 11 < 2) {
      return 0;
    }
    return 11 - (sum % 11);
  }

  buildString() {
    let CPF = "";

    this.buildCPF.splice(3, 0, ".");
    this.buildCPF.splice(7, 0, ".");
    this.buildCPF.splice(11, 0, "-");

    for (let i = 0; i < this.buildCPF.length; i++) {
      CPF += this.buildCPF[i];
    }

    return CPF;
  }
}

File: C/src/GeneratorAlgorithm.c
# include <stdio.h>
# include <stdlib.h>
# include <time.h>

void generateCPF(int buildCPF[]);
int digit(int buildCPF[], int index);
int verifySum(int sum);
void buildString(int buildCPF[]);

int main() {
  int buildCPF[11];
  srand(time(NULL));

  generateCPF(buildCPF);

  return 0;
}

void generateCPF(int buildCPF[]) {
  int index = 10;

  for (int i = 0; i < 9; i++) {
    buildCPF[i] = rand() % 10;
  }

  for (int i = 9; i < 11; i++) {
    buildCPF[i] = digit(buildCPF, index);
    index++;
  }

  buildString(buildCPF);
}

int digit(int buildCPF[], int index) {
  int sum = 0, length = index - 1;

  for (int i = 0; i < length; i++) {
    sum += buildCPF[i] * index;
    index--;
  }

  return verifySum(sum);
}

int verifySum(int sum) {
  if ((sum % 11) < 2)
    return 0;
  return 11 - (sum % 11);
}

void buildString(int buildCPF[]) {
  char cpf[14];

  for (int i = 0; i < 11; i++) {
    cpf[i] = buildCPF[i] + '0';
  }

  for (int i = 13; i >= 3; i--) {
    cpf[i + 1] = cpf[i];
  }

  for (int i = 13; i >= 7; i--) {
    cpf[i + 1] = cpf[i];
  }

  for (int i = 13; i >= 11; i--) {
    cpf[i + 1] = cpf[i];
  }

  cpf[3] = '.';
  cpf[7] = '.';
  cpf[11] = '-';

  printf("%s", cpf);
}
